tomcat/src/main/java/org/apache/coyote/http11/request/HttpRequest.java

```diff
@@ -0,0 +1,78 @@
+package org.apache.coyote.http11.request;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class HttpRequest {
+
+    private final RequestLine requestLine;
+    private final RequestHeaders requestHeaders;
+    private final RequestBody requestBody;
+
+    private HttpRequest(RequestLine requestLine, RequestHeaders requestHeaders, RequestBody requestBody) {
+        this.requestLine = requestLine;
+        this.requestHeaders = requestHeaders;
+        this.requestBody = requestBody;
+    }
+
+    public static HttpRequest from(BufferedReader bufferedReader) throws IOException {
+        RequestLine requestLine = RequestLine.from(bufferedReader.readLine());
+        RequestHeaders requestHeaders = RequestHeaders.from(extractHeaders(bufferedReader));
+        RequestBody requestBody = RequestBody.from(extractBody(bufferedReader, requestHeaders));
+        return new HttpRequest(requestLine, requestHeaders, requestBody);
+    }
+
+    private static List<String> extractHeaders(BufferedReader bufferedReader) throws IOException {
+        List<String> headers = new ArrayList<>();
+        while (true) {
+            String line = bufferedReader.readLine();
+            if (line == null || line.isBlank()) {
+                break;
+            }
+            headers.add(line);
+        }
```

Comment:
```suggestion
        String line;
        while (!(line = bufferedReader.readLine()).isBlank()) {
            headers.add(line);
        }
```
r: 이렇게 하면 indent를 줄일 수 있을 것 같아요! 그리고 `isBlank` 메서드에서 null까지 검사해주기 때문에 둘 다 검사할 필요는 없어보이네요.

Reply:
오오 감사합니다!😊